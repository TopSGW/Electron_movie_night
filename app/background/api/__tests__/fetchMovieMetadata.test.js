'use strict'
/* globals describe, test, expect, jest */

// Mock the API responses.
jest.mock('../../../shared/request', () => {
  return {
    getJSON: jest.fn((url, validator) => {
      if (url.includes('reject')) {
        return Promise.reject(new Error())
      }

      if (url.includes('error')) {
        validator({ error: 'throws' })
      } else {
        validator({ url })
      }

      let metadata = {}
      metadata.successQuery = url.slice(url.indexOf('file='))
      if (url.includes('imgUrl')) {
        metadata.imgUrl = 'http://poster.png'
        metadata.imdbID = 'tt123'
      }
      return Promise.resolve(metadata)
    })
  }
})

// Mock the poster image path returned from config so snapshots
// do not contain any host information.
jest.mock('../../../../config', () => {
  return {
    posterImagePath: '/path/to/movie/posters'
  }
})

// This is the module we are testing
const { fetchMovieMetadata } = require('../fetchMovieMetadata')

describe('fetchMovieMetadata', () => {
  test('resolves with fileInfo metadata on successful fetch', () => {
    const movieFile = 'OldSchool.avi'
    return expect(fetchMovieMetadata(movieFile))
      .resolves.toMatchSnapshot()
  })

  test('resolves with imgFile when metadata contains imgUrl', () => {
    const movieFile = 'OldSchool imgUrl'
    return expect(fetchMovieMetadata(movieFile))
      .resolves.toMatchSnapshot()
  })

  test('rejects on fetch error', () => {
    const movieFile = 'reject'
    return expect(fetchMovieMetadata(movieFile))
      .rejects.toBeInstanceOf(Error)
  })

  test('throws on data error', () => {
    expect.assertions(1)

    const movieFile = 'error'
    try {
      fetchMovieMetadata(movieFile)
    } catch (err) {
      expect(err).toBeInstanceOf(Error)
    }
  })
})
