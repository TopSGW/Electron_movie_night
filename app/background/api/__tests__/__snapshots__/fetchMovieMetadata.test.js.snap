// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetchMovieMetadata resolves with fileInfo metadata on successful fetch 1`] = `
Object {
  "imgFile": "",
  "location": "OldSchool.avi",
  "query": "file=OldSchool.avi",
  "successQuery": "file=OldSchool.avi",
}
`;

exports[`fetchMovieMetadata resolves with imgFile when metadata contains imgUrl 1`] = `
Object {
  "imdbID": "tt123",
  "imgFile": "/path/to/movie/posters/tt123.png",
  "imgUrl": "http://poster.png",
  "location": "OldSchool imgUrl",
  "query": "file=OldSchool%20imgUrl",
  "successQuery": "file=OldSchool%20imgUrl",
}
`;
